docker/docker_client.go

```diff
@@ -1,18 +1,43 @@
 package docker
 
 import (
+	"context"
+	"fmt"
 	"net/http"
 	"net/url"
 	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"time"
 
 	"knative.dev/kn-plugin-func/ssh"
 
 	"github.com/docker/docker/client"
 )
 
-func NewDockerClient() (dockerClient client.CommonAPIClient, dockerHost string, err error) {
+func NewDockerClient(defaultDockerHost string) (dockerClient client.CommonAPIClient, dockerHost string, err error) {
```

Comment:
Minor nit, but since this is in the `docker` package, this could just be `docker.NewClient` and even `defaultHost` etc.